### PR TITLE
Controller mapping profile and controller attribute fixes

### PR DESCRIPTION
**Pull request by davidkline-ms**
Overview
---
- Wrap editor specific code in the controller mapping profile in #if / #endif.
- Add "using Microsoft.MixedReality.Toolkit.Core.Extensions" when building UWP on .net backend

This does not 100% resolve .net backend build breaks. There are some changes needed in interactables to work around AppDomain usage

### CODE REVIEW

File: Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerMappingProfile.cs [hunk 1]
```diff
@@ -19,24 +19,25 @@ namespace Microsoft.MixedReality.Toolkit.Core.Definitions.Devices
...
-
-        public static Type[] CustomControllerMappingTypes { get => (from type in ControllerMappingTypes where UsesCustomInteractionMapping(type) select type).ToArray(); }
-
         [SerializeField]
         [Tooltip("The list of controller templates your application can use.")]
         private MixedRealityControllerMapping[] mixedRealityControllerMappingProfiles = new MixedRealityControllerMapping[0];
 
         public MixedRealityControllerMapping[] MixedRealityControllerMappingProfiles => mixedRealityControllerMappingProfiles;
 
+#if UNITY_EDITOR
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerMappingProfile.cs]: Have we confirmed a .NET build works and deploys without this section in the build on both Windows and Standalone @davidkline-ms ?

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityControllerMappingProfile.cs]: I have confirmed an il2pp build deploys and runs. there are other parts (interactables) of the project that are preventing successful .net backend builds.

**Comment by davidkline-ms** [comment]
moving to target mrtk_developent to give a bit more time to verify all build targets (and ideally get the interactable changes in place as well)

File: Assets/MixedRealityToolkit/Attributes/MixedRealityControllerAttribute.cs [hunk 2]
```diff
@@ -6,6 +6,10 @@
 using System;
 using System.Linq;
 
+#if WINDOWS_UWP && !ENABLE_IL2CPP
```
Comment (chrisfromwork) [inline Assets/MixedRealityToolkit/Attributes/MixedRealityControllerAttribute.cs]: did we get any comments in ship room whether we would support .net? From what i understand newer versions of unity don't provide .net tooling?

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Attributes/MixedRealityControllerAttribute.cs]: The standing policy is to support the backends that Unity supports. 2018.4 will continue to support the .NET backend for 2 years following its release.

As we move to Unity 2019, we will only support il2cpp. For as long as we support Unity 2018, we will also support .NET.

Comment (keveleigh) [reply Assets/MixedRealityToolkit/Attributes/MixedRealityControllerAttribute.cs]: Could this be simplified to just `#if ENABLE_DOTNET`?

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Attributes/MixedRealityControllerAttribute.cs]: I wasn't aware of that define! We should create an issue to replace all of the  #if WINDOWS_UWP && !ENABLE_IL2CPP instances with it!

**Pull request merged**